index.js

```diff
@@ -99,22 +113,27 @@ module.exports = function(source, map) {
 	const isServer = this.target === 'node' || (options.generate && options.generate == 'ssr');
 	const isProduction = this.minimize || process.env.NODE_ENV === 'production';
 
-	options.filename = this.resourcePath;
-	if (!('format' in options)) {
-		options.format = major_version >= 3 ? 'esm' : 'es';
+	const compileOptions = {
+		filename: this.resourcePath,
+		format: options.format || (major_version >= 3 ? 'esm' : 'es')
+	};
+
+	if (major_version >= 3) {
+		// TODO anything?
+	} else {
+		compileOptions.shared = options.shared || 'svelte/shared.js';
+		compileOptions.name = capitalize(sanitize(options.filename));
 	}
-	if (!('shared' in options)) {
-		const shared = (major_version >= 3 ? 'svelte/internal.js' : 'svelte/shared.js');
-		options.shared = (options.format === 'es' || options.format === 'esm') &&
-			requireRelative.resolve(shared, process.cwd());
+
+	for (const option in options) {
+		if (!pluginOptions[option]) compileOptions[option] = options[option];
 	}
-	if (!('name' in options)) options.name = capitalize(sanitize(options.filename));
-	if (!('onwarn' in options)) options.onwarn = warning => this.emitWarning(new Error(warning));
-	if (options.emitCss) options.css = false;
-	if (options.externalDependencies) options.externalDependencies.forEach(dep => this.addDependency(dep));
+
+	// if (!('onwarn' in options)) options.onwarn = warning => this.emitWarning(new Error(warning));
```

Comment:
I haven't looked at these changes really, but having this code in here commented out seems like bad form.

Reply:
oh right, yeah